xcdat/temporal.py

```diff
@@ -150,6 +150,17 @@ def __init__(self, dataset: xr.Dataset):
 
         self._time_bounds = self._dataset.bounds.get_bounds("T").copy()
 
+        try:
+            self.calendar = self._dataset[self._dim].encoding["calendar"]
+            self.date_type = get_date_type(self.calendar)
+        except KeyError:
+            raise KeyError(
+                "This dataset's time coordinates do not have a 'calendar' encoding "
+                "attribute set, which might indicate that the time coordinates were not "
+                "decoded to datetime objects. Ensure that the time coordinates are "
+                "decoded before performing temporal averaging operations."
+            )
```

Comment:
We now consider the "calendar" attribute and the associated `cftime` date type object for temporal averaging operations.

Reply:
Should "were not decoded to datetime objects" be "were not decoded to cftime.datetime objects"? Or are they sufficiently interchangeable?

Reply:
Temporal averaging accepts both `datetime.datetime` and `cftime.datetime` objects, but the end result will always be `cftime.datetime`.

I think it is sufficiently interchangeable to reference both as just `datetime`.